crates/bevy_ecs/src/query/filter.rs

```diff
@@ -588,52 +574,81 @@ macro_rules! impl_tick_filter {
             const IS_ARCHETYPAL:  bool = false;
 
             unsafe fn set_table(&mut self, state: &Self::State, table: &'w Table) {
-                self.table_ticks = Some(table.get_column(state.component_id).unwrap().get_ticks_slice().into());
+                match T::Storage::STORAGE_TYPE {
```

Comment:
unnecessary match

Reply:
We might get other storage types in the future, right? In that case using a match ensures that this match is updated if necessary.

Reply:
That's true, if we opt to keep this instead of removing we should atleast replace the `Sparse` arm with an `unreachable` or `debug_checked_unreachable` so that we get a panic or miri CI failure if things go wrong instead of silently doing nothing

Reply:
Should I go ahead and add a debug_unreachable check in each of these arms then? Or just remove the match altogether?

I tried to merge `set_archetype` and `set_table` together, since both are doing the same thing in most of the impls with the sparse arm empty, but it snagged on the Option<T> based ones requiring different matches.

Reply:
Done.